application/backend/src/main/java/cmpe451/group12/beabee/goalspace/controller/GroupGoalController.java

```diff
@@ -0,0 +1,137 @@
+    @PostMapping("/{groupgoal_id}/token")
+    public MessageResponse regenerateToken(@PathVariable @ApiParam(value = "Id of the group goal", example = "5") Long groupgoal_id) {
+        return groupGoalService.regenerateToken(groupgoal_id);
+    }
+
+    @ApiOperation(value = "Update a group goal.")
+    @PutMapping("")
+    public MessageResponse updateAGroupgoal(@RequestBody @ApiParam(
+            value = "A JSON value representing a transaction. An example of the expected schema can be found down here.",
+            examples = @Example(value =
+            @ExampleProperty(
+                    value = "{\n" +
+                            "  \"createdAt*\": \"2021-11-20T10:01:10.538Z\",\n" +
+                            "  \"deadline*\": \"2021-11-20T10:01:10.538Z\",\n" +
+                            "  \"description*\": \"string\",\n" +
+                            "  \"goalType\": \"GROUPGOAL\",\n" +
+                            "  \"id*\": 0,\n" +
+                            "  \"isDone*\": true,\n" +
+                            "  \"title*\": \"string\"\n" +
+                            "}"
+            )
+            ) ) GroupGoalGetDto groupGoalGetDto){
+        return groupGoalService.updateAGroupgoal(groupGoalGetDto);
+    }
+
+    @ApiOperation(value = "Create a group goal.")
+    @PostMapping("/{user_id}")
+    public MessageResponse createAGroupgoal(@RequestBody @ApiParam(
+            value = "A JSON value representing a transaction. An example of the expected schema can be found down here.",
+            examples = @Example(value =
+            @ExampleProperty(
+                    value = "{\n" +
+                            "  \"deadline*\": \"2021-11-20T10:01:10.538Z\",\n" +
+                            "  \"description*\": \"string\",\n" +
+                            "  \"title*\": \"string\"\n" +
+                            "}"
+            )
+            ) ) GroupGoalPostDTO groupGoalPostDTO, @PathVariable @ApiParam(value = "Id of the user.",example = "5")Long user_id){
+        return groupGoalService.createAGroupgoal(user_id, groupGoalPostDTO);
+    }
+
+    @ApiOperation(value = "Delete a goal.")
+    @DeleteMapping("/{groupgoal_id}")
+    public MessageResponse deleteGroupgoal(@PathVariable @ApiParam(value = "Id of the group goal.",example = "5")Long groupgoal_id){
+        return groupGoalService.deleteGroupgoal(groupgoal_id);
+    }
+
+    @ApiOperation(value = "Join a group goal with token")
+    @PostMapping("/{user_id}/{token}")
+    public GroupGoalDTOShort joinWithToken(
+            @PathVariable @ApiParam(value = "Id of the user", example = "5") Long user_id,
+            @PathVariable @ApiParam(value = "Token of the group", example = "5wbwf6yUxVBcr48AMbz9cb") String token){
+        return groupGoalService.joinWithToken(user_id, token);
+    }
+
+    @ApiOperation(value = "Add a new member to the Group Goal")
+    @PostMapping("/{groupgoal_id}/{username}")
+    public MessageResponse addMember(
```

Comment:
may not be needed since joinWithToken() is going to be used while joining a group.